src/index.ts

```diff
@@ -82,13 +83,19 @@ class TCP implements Transport {
 
     const maConn = toMultiaddrConnection(socket, {
       remoteAddr: ma,
-      signal: options.signal,
       socketInactivityTimeout: this.opts.outboundSocketInactivityTimeout,
       socketCloseTimeout: this.opts.socketCloseTimeout
     })
+
+    const onAbort = () => socket.end()
```

Comment:
On dial abort should .end (send FIN packet), .destroy directly, or maConn.close?